coala_quickstart/interaction/Logo.py

```diff
@@ -0,0 +1,46 @@
+from coala_quickstart.Strings import coala_bear_logo, welcome_messages
+
+
+def print_logo(printer):
+    """
+    Prints the coala bear logo alongside a welcome message.
+
+    :param printer:
+        A ``ConsolePrinter`` object used for console interaction.
+    """
+    max_length = len(max(coala_bear_logo, key=len))
+    start_line = 2
+
+    current_line = 0
+    current_para = 0
+    current_word = 0
+
+    paragraphs = []
+    for welcome_message in welcome_messages:
+        paragraphs.append(welcome_message.split(" "))
+        paragraphs.append([""])
+
+    for bear_line in coala_bear_logo:
+        printer.print(bear_line, color="white", end="")
+
+        if current_line >= start_line and current_para < len(paragraphs):
+            words = paragraphs[current_para]
+            if current_word < len(words):
+                padding = max_length - len(bear_line) + 2
+                printed_length = max_length + 2
+                printer.print(" " * padding, color="white", end="")
+
+                while printed_length < 79 and current_word < len(words):
+                    printer.print(
+                        words[current_word],
+                        color="blue",
+                        end=" ")
+                    printed_length += len(words[current_word]) + 1
+                    current_word += 1
+
+                if current_word == len(words):
+                    current_word = 0
+                    current_para += 1
+
+        printer.print("", color="blue")
+        current_line += 1
```

Comment:
wohoooo! This is complicated. You can copy stuff from https://gitlab.com/sims1253/LoneCoffin/blob/master/utils.py#L1 it's my code and you have my consent to reuse it under AGPL.